src/typechecker.erl

```diff
@@ -1582,13 +1582,18 @@ do_type_check_expr(Env, {op, P, RelOp, Arg1, Arg2}) when
       (RelOp == '>')  or (RelOp == '<') ->
     type_check_rel_op(Env, RelOp, P, Arg1, Arg2);
 do_type_check_expr(Env, {op, P, Op, Arg1, Arg2}) when
-      Op == '+' orelse Op == '-' orelse Op == '*' orelse Op == '/' ->
-    type_check_arith_op(Env, Op, P, Arg1, Arg2);
-do_type_check_expr(Env, {op, P, Op, Arg1, Arg2}) when
+      Op == '+' orelse Op == '-' orelse Op == '*' orelse Op == '/' orelse
       Op == 'div'  orelse Op == 'rem' orelse
       Op == 'band' orelse Op == 'bor' orelse Op == 'bxor' orelse
-      Op == 'bsl'  orelse Op == 'bsr' ->
-    type_check_int_op(Env, Op, P, Arg1, Arg2);
+      Op == 'bsl'  orelse Op == 'bsr'->
+    FunTy = case gradualizer_db:get_spec(erlang, Op, 2) of
```

Comment:
superfluous case? could be just matching?

Reply:
Absolutely! Will fix while working on getting this to a mergeable state!